### PR TITLE
Implement POST Method for /data endpoint

### PR DESCRIPTION
**Pull request by samuelsogbesan**
This PR is designed to integrate the POST Method handler on the /data servlet.

- **Add Key Dependency (GSON)**
- **Create container class for FormData**
- **Parse request data and encapsulate data inside object**
- **Repackage information as JSON (using GSON) and send in response**

A future pull request will implement the datastore and update this method to store the encapsulated information there.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -14,19 +14,49 @@
...
+import java.util.Arrays;
+import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+/** Container class for Form data */
+class FormData {
+  private String nickname; //The alias of the commenter
```
Comment (krasimirgg) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: [nit: make sure there is 1 space between `//` and the first letter of a comment](https://google.github.io/styleguide/javaguide.html#s4.8.6-comments)

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -14,19 +14,49 @@
...
+  private String comment; // The assocaited comment
```
Comment (krasimirgg) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: typo: `s/assocaited/associated/` (replace assocaited with associated)

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -14,19 +14,49 @@
...
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
     response.getWriter().println("Hello Sam");
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String nickname = getParameter(request, "nickname", "anon");
+    String comment = getParameter(request, "comment", "");
+    List<FormData> formdata = Arrays.asList(new FormData(nickname,comment));
```
Comment (krasimirgg) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: nit: add a space between `,` and `comment`

Comment (krasimirgg) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: nit: use camelCase for the `formdata` variable name (`formData`).

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -14,19 +14,49 @@
...
+    response.setContentType("application/json;");
+    response.getWriter().println(new Gson().toJson(formdata));
+  }
+
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
```
Comment (krasimirgg) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:55]: it's great that you extracted this into a helper function!

**Review by krasimirgg — CHANGES_REQUESTED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (samuelsogbesan) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Updated in commit 4211bf8

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (samuelsogbesan) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Updated in commit e9d80a6

Comment (samuelsogbesan) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Updated with commit 7013e09

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (samuelsogbesan) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Updated with commit 9910c92

**Review by krasimirgg — APPROVED** [review]
Great!

**Pull request merged**